Sources/MapboxCoreNavigation/NavSDK.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+import MapboxDirections
+
+public struct NavSDK {
+    private static let lock: NSLock = .init()
+    private static var isInitialized: Bool = false
+    public static var directions: Directions!
+    public static var tileStoreConfiguration: TileStoreConfiguration!
+
+    public static func initialize(directions: Directions = .shared,
+                                  tileStoreConfiguration: TileStoreConfiguration = .default) {
```

Comment:
Not too keen on an initializer that has to be called before everything else in the application. Developers would be prone to forgetting to call it, and overall it feels like a non-Cocoa way of setting up a library versus something more tightly scoped. If this code really has to run as soon as possible, then consider sticking an Objective-C `+initialize` method in CMapboxCoreNavigation.

Reply:
As per discussion, this will become an advance method with fallback to defaults.

Reply:
This is no longer a required call. Please review the latest revision.